packages/dds/tree/src/feature-libraries/persisted-config/persistedConfig.ts

```diff
@@ -0,0 +1,336 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+
+// TODO: Document, decide if semver-like even provides value here (over e.g. plain-old number).
+export type Version = `${number}.${number}.${number}${"-internal" | ""}`;
+
+export interface PersistedConfig {
+	version: Version;
+	// Strictly increasing number which identifies the iteration of the proposed protocol. It's suggested
+	// that applications are able to modify this using the same mechanisms with which they can modify a DDS's
+	// configuration, which enables safe rollback policies.
+	// Example: application enables feature flag 1 in iteration 2, then realizes there is a problem with it.
+	// application then disables feature flag 1 in iteration 3.
+	// Using a single version number and monotonic increase on it isn't sufficient: DDSes may support configuration
+	// which impacts their serialization format with independent semantics (ex: attribution, policies to include history)
+	protocolIteration: number;
+	flags: Record<string, boolean>; // TODO: consider making this generic
```

Comment:
> Additionally, when adding a new kind of configuration support its default is really important since you want to have apps that don't support it and apps that do support it correctly interop when the option has never been set (and thus is at its default).

> Might be better to make them a number instead

yeah, fair point. I think I had planned on making the config optional, which could work but isn't great (leaks the fact that there's back-compat concerns with the set of configuration that one can specify to the public API). I think the two above concerns can be unified with the convention that "feature flags" always add data to the summary and are introduced at version 0 (which is off). so apps always tolerate seeing a config with feature flag 0 they don't have metadata for, but throw upon seeing versions greater than their max supported one. This also would enable natural support for multiple supported persisted formats for optional features like attribution data or whether summary history should be written.